CSharpSourceCode/Battle/CrosshairMissionBehavior/CustomCrosshairMissionBehavior.cs

```diff
@@ -140,18 +140,37 @@ private void UpdateAbilityCrosshairVisibility()
             {
                 if (!abilityComponent.CurrentAbility.CanCast(Agent.Main))
                 {
-                    if (abilityCrosshair.IsVisible)
-                        abilityCrosshair.Hide();
+                    abilityCrosshair.Hide();
                 }
                 else
                 {
-                    abilityCrosshair.Tick();
-                    if (!abilityCrosshair.IsVisible)
+                    if (abilityCrosshair.CrosshairType == CrosshairType.CenteredAOE || IsCorrectDirection())
+                    {
+                        abilityCrosshair.Tick();
                         abilityCrosshair.Show();
+                    }
+                    else
+                    {
+                        abilityCrosshair.Hide();
+                    }
                 }
             }
         }
 
+        private bool IsCorrectDirection()
+        {
+            var frame1 = Agent.Main.LookFrame;
+            frame1.rotation.f.z = 0;
+            frame1.Advance(1);
+
+            var frame2 = Agent.Main.LookFrame;
+            frame2.rotation.f = Agent.Main.MountAgent.GetMovementDirection();
+            frame2.rotation.f.z = 0;
+            frame2.Advance(1);
+
+            return frame1.origin.Distance(frame2.origin) < 1.2f;
```

Comment:
Sorry if I might have mistaken something, but the math looks wrong to me. Refractor this to a method called "CalculateAngleBetweenVectors" or similar. inside this you might then make a proper calculation of the vector. like here in  https://www.omnicalculator.com/math/angle-between-two-vectors

Reply:
Yep, i created this solution quickly and not so properly if talking about formula and naming. Thanks for your advice, i'll use it.